<a href="https://colab.research.google.com/github/unnati-m/Google_Collab/blob/master/assign_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1.Write a program that reads an unspecified number of integers, determines how many positive and negative values have been read, and computes the total and average of the input values (not counting zeros). Your program ends with the input 0. Display the average as a floating-point number.

In [4]:
nums = list(map(int , input("Enter the numbers: ").split()))
pos,neg=0,0
total,avg=0,0
for x in nums:
  if x>=0:
    pos+=1
  else:
    neg+=1
  total=total+x
avg=total/len(nums)
print("Positive integers are:",pos)
print("Negative integers are:",neg)
print("total is:",total)
print("Average is:",avg)

Enter the numbers: 3 4 5 -3 -1 0
Positive integers are: 4
Negative integers are: 2
total is: 8
Average is: 1.3333333333333333


2.(Financial application: compute future tuition) Suppose that the tuition for a university is $10,000 this year and increases 5% every year. Write a program that computes the tuition in ten years and the total cost of four years’ worth of tuition starting ten years from now.

In [11]:
x = 10000
y = 10

while(y):
  res = (x/100)*5
  x += res
  y -= 1

print(f'Tuition in 10 years: {round(x)}')

fnl = x
g = 3
while(g):
  res = (x/100)*5
  x += res
  fnl += x
  g -= 1

print(f'Tuition for 4 years starting 10 years from now: {fnl}')

Tuition in 10 years: 16289
Tuition for 4 years starting 10 years from now: 70207.39453239119


3.(Find the highest score) Write a program that prompts the user to enter the number of students and each student’s score, and displays the highest score.

In [12]:
n = int(input('Enter the number of students: '))
l = []
for i in range(n):
  s = int(input(f'Enter score of student {i+1}: '))
  l.append(s)
h=0
for i in l:
  if h<i:
    h=i

print(f'Highest score is: {h}')

Enter the number of students: 5
Enter score of student 1: 67
Enter score of student 2: 78
Enter score of student 3: 63
Enter score of student 4: 82
Enter score of student 5: 89
Highest score is: 89


4.(Find the two highest scores) Write a program that prompts the user to enter the number of students and each student’s score, and displays the highest and secondhighest scores.

In [13]:
n = int(input('Enter the number of students: '))
l = []
for i in range(n):
  s = int(input(f'Enter score of student {i+1}: '))
  l.append(s)
h=0
sc = 0
for i in l:
  if h<i:
    sh = h
    h=i

print(f'Highest score is: {h} and second-highest score is {sh}')

Enter the number of students: 5
Enter score of student 1: 77
Enter score of student 2: 65
Enter score of student 3: 81
Enter score of student 4: 86
Enter score of student 5: 90
Highest score is: 90 and second-highest score is 86


5.(Find numbers divisible by 5 and 6) Write a program that displays, ten numbers per line, all the numbers from 100 to 1,000 that are divisible by 5 and 6. The numbers are separated by exactly one space.

In [14]:
for i in range(100, 1001):
  if i%5==0 or i%6==0:
    print(i, end = ' ')

100 102 105 108 110 114 115 120 125 126 130 132 135 138 140 144 145 150 155 156 160 162 165 168 170 174 175 180 185 186 190 192 195 198 200 204 205 210 215 216 220 222 225 228 230 234 235 240 245 246 250 252 255 258 260 264 265 270 275 276 280 282 285 288 290 294 295 300 305 306 310 312 315 318 320 324 325 330 335 336 340 342 345 348 350 354 355 360 365 366 370 372 375 378 380 384 385 390 395 396 400 402 405 408 410 414 415 420 425 426 430 432 435 438 440 444 445 450 455 456 460 462 465 468 470 474 475 480 485 486 490 492 495 498 500 504 505 510 515 516 520 522 525 528 530 534 535 540 545 546 550 552 555 558 560 564 565 570 575 576 580 582 585 588 590 594 595 600 605 606 610 612 615 618 620 624 625 630 635 636 640 642 645 648 650 654 655 660 665 666 670 672 675 678 680 684 685 690 695 696 700 702 705 708 710 714 715 720 725 726 730 732 735 738 740 744 745 750 755 756 760 762 765 768 770 774 775 780 785 786 790 792 795 798 800 804 805 810 815 816 820 822 825 828 830 834 835 840 845 846 

6.Write a program that reads an integer and displays all its smallest factors, also known as prime factors. For example, if the input integer is 120, the output should be as follows: 2, 2, 2, 3, 5

In [15]:
n = int(input("Enter a number: "))
i = 2
while i<n+1:
  if n%i == 0:
    print(i, end = ', ')
    n = n/i
  else:
    i += 1

Enter a number: 46
2, 23, 

7.Write a program to prints all letters except ‘y' and 't' in the string "Basic python Programming" using continue statement.

In [16]:
st = 'Basic python Programming'
for x in st:
  l = ""
  for y in x:
    if(y!='y' and y!='t'):
      l+=y
    else:
      continue
  print(l, end="")

Basic phon Programming

8.Write a program to prints letters in the string "Basic python Programming" except "g" using break statement.

In [17]:
s = 'Basic python Programming'
for i in s:
  if i == 'g':
    break
  else:
    print(i, end = '')

Basic python Pro

9.Write a program to print table of given number using while loop

In [18]:
n = int(input('Enter a number: '))
i = 1
while i<11:
  print(f'{n} * {i} = {n*i}')
  i += 1

Enter a number: 24
24 * 1 = 24
24 * 2 = 48
24 * 3 = 72
24 * 4 = 96
24 * 5 = 120
24 * 6 = 144
24 * 7 = 168
24 * 8 = 192
24 * 9 = 216
24 * 10 = 240


10.Write a program to print the following pattern using for loops.
1
11
111
1111
11111

In [19]:
for i in range(6):
  print('1'*i)


1
11
111
1111
11111


11.Write a program to print the following pattern using for loops.
    *
   * *
  * * *
 * * * *
* * * * *

In [20]:
for i in range(1,6):
  print(' '*abs(i-5),end = '')
  print('* '*i)

    * 
   * * 
  * * * 
 * * * * 
* * * * * 


In [ ]:
12.Write a program to check whether the given number is a Palindrome Number using Function. (use looping stmt)

In [21]:
def checkPalindrome(n):
  temp=n
  rev=0
  while(n>0):
      dig=n%10
      rev=rev*10+dig
      n=n//10
  if(temp==rev):
      print("The number is a palindrome!")
  else:
      print("The number isn't a palindrome!")

n=int(input("Enter number: "))
checkPalindrome(n)

Enter number: 454
The number is a palindrome!
